website/i18n/zh/docusaurus-plugin-content-blog/2021/12/01/release-apache-apisix-2.11.md

```diff
@@ -0,0 +1,176 @@
+    url: "https://github.com/SylviaBABY"
+    image_url: "https://avatars.githubusercontent.com/u/39793568?v=4"
+keywords: 
+- Apache APISIX
+- 版本发布
+- WASM
+- Azure
+- LDAP
+description: Apache APISIX 2.11.0 版本正式发布，这是继 2.10.0 LTS 版本后第一版带有新功能的版本。
+tags: [Release]
+---
+
+> Apache APISIX 2.11.0 版本正式发布，这是继 2.10.0 LTS 版本后第一版带有新功能的版本。
+
+<!--truncate-->
+
+Apache APISIX 2.11.0 版本是继上次 2.10.0 LTS 版本发布后，第一个带有新功能的版本。不仅丰富了插件库，还带来了新鲜的生态支持。
+
+## 新功能：新增基于 LDAP 的认证插件
+
+Apache APISIX 长长的认证插件名单上又添加了新成员——基于 LDAP 的 `ldap-auth` 插件。通过该插件我们可以打通 LDAP 的账户体系和 Apache APISIX 的 Consumer 机制。
+
+我们通过代码端简单进行示例展示：
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/consumers -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "username": "user01",
+    "plugins": {
+        "ldap-auth": {
+            "user_dn": "cn=user01,ou=users,dc=example,dc=org"
+        }
+    }
+}'
+```
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "methods": ["GET"],
+    "uri": "/hello",
+    "plugins": {
+        "ldap-auth": {
+            "base_dn": "ou=users,dc=example,dc=org",
+            "ldap_uri": "localhost:1389",
+            "uid": "cn"
+        },
+    },
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+```
+
+在上述配置中，我们把 `user01` 绑定到 `route 1` 上。这样我们在访问 `route 1` 时就带上了 `user01` 的密码，就可以通过 LDAP 的身份认证了。
+
+运行结果就像这样：
+
+```shell
+curl -i -uuser01:password1 http://127.0.0.1:9080/hello
+HTTP/1.1 200 OK
+...
+hello, world
+```
+
+## 新功能：可观测性层面对接更多监控体系
+
+新版本的 Apache APISIX 丰富了对外部监控服务的支持。在这方面，我们新增了两个插件：
+
+* 上报指标到 datadog 的 `datadog` 插件
+* 上报访问日志到 Apache Skywalking 的 `skywalking-logger` 插件
+
+Datadog 是海外广泛使用的 SaaS 监控服务，而 Apache Skywalking 是享誉全球的开源监控软件。如今用户只需在路由上简单配置一下，就能实现与它们的对接。
+
+Datadog 示例：
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+      "plugins": {
+            "datadog": {}
+       },
+      "upstream": {
+           "type": "roundrobin",
+           "nodes": {
+               "127.0.0.1:1980": 1
+           }
+      },
+      "uri": "/hello"
+}'
+```
+
+Apache SkyWalking 示例：
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "plugins":{
+        "skywalking-logger":{
+            "endpoint_addr":"http://127.0.0.1:12800"
+        }
+    },
+    "upstream":{
+        "type":"roundrobin",
+        "nodes":{
+            "127.0.0.1:1980":1
+        }
+    },
+    "uri":"/hello"
+}'
+```
+
+## 新功能：通过网关暴露 Azure 的 FaaS 函数
+
+网关能做的不仅仅是代理内部服务，我们也可以用它来连接外部的系统。
+
+现在通过 `azure-functions` 插件，就可以利用 HTTP 请求触发 Azure functions 服务上的函数了。
+
+接下来我们将通过下方示例来展示如何把 Azure 上配置好的函数，跟 Apache APISIX 上 `/azure_HttpTrigger` 路由连接起来。
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "plugins":{
```

Comment:
Would better use 2 spaces for JSON.

Reply:
The code that I copied directly from the original text.
I'll pay attention to it later.